packages/ui/src/domainModules/collectionMammals/components/MammalManager/MainColumn/RecordForm/formParts/CollectingLocality/CollectingPosition.js

```diff
@@ -0,0 +1,189 @@
+  minimumElevationInMeters: undefined,
+  onEdit: undefined,
+  uncertaintyInMeters: undefined,
+}
+
+const getCoordinates = (latitude, longitude) => {
+  const latNumber = latitude && Number(latitude)
+  const lat = latNumber < 0 ? 'S' : 'N'
+
+  const lonNumber = longitude && Number(longitude)
+  const lon = longitude < 0 ? 'W' : 'E'
+
+  if (latitude && longitude) {
+    return `${Math.abs(latNumber)} ${lat}, ${Math.abs(lonNumber)} 
+      ${lon}, (WGS84) `
+  }
+  if (latitude) {
+    return `${Math.abs(latNumber)} ${lat}, (WGS84)`
+  }
+  return `${Math.abs(lonNumber)} ${lon}, (WGS84)`
+}
+
+const getElevation = (maximumElevationInMeters, minimumElevationInMeters) => {
+  if (maximumElevationInMeters === undefined) {
+    return `Elevation: ${minimumElevationInMeters} m`
+  }
+
+  if (minimumElevationInMeters === undefined) {
+    return `Elevation: ${maximumElevationInMeters} m`
+  }
+  return `Elevation: ${minimumElevationInMeters} to
+    ${maximumElevationInMeters} m`
+}
+
+const getDepth = (maximumDepthInMeters, minimumDepthInMeters) => {
+  if (maximumDepthInMeters === undefined) {
+    return `Depth: ${minimumDepthInMeters} m`
+  }
+
+  if (!minimumDepthInMeters === undefined) {
+    return `Depth: ${minimumDepthInMeters} m`
+  }
+  return `Depth: ${minimumDepthInMeters} to ${maximumDepthInMeters} m`
+}
+
+const buildCollectingPosition = (...values) => {
+  return values.filter(Boolean).join(', ')
+}
+
+const mapStateToProps = (state, { formValueSelector, getPath }) => {
+  return {
+    coordinatesVerbatim: formValueSelector(
+      state,
+      getPath('coordinatesVerbatim')
+    ),
+    georeferenceSourcesText: formValueSelector(
+      state,
+      getPath('georeferenceSourcesText')
+    ),
+    latitude: formValueSelector(state, getPath('position.latitude')),
+    longitude: formValueSelector(state, getPath('position.longitude')),
+    maximumDepthInMeters: formValueSelector(
+      state,
+      getPath('verticalPosition.maximumDepthInMeters')
+    ),
+    maximumElevationInMeters: formValueSelector(
+      state,
+      getPath('verticalPosition.maximumElevationInMeters')
+    ),
+    minimumDepthInMeters: formValueSelector(
+      state,
+      getPath('verticalPosition.minimumDepthInMeters')
+    ),
+    minimumElevationInMeters: formValueSelector(
+      state,
+      getPath('verticalPosition.minimumElevationInMeters')
+    ),
+    uncertaintyInMeters: formValueSelector(
+      state,
+      getPath('position.uncertaintyInMeters')
+    ),
+  }
+}
+
+class CollectingPosition extends PureComponent {
+  render() {
+    const {
+      coordinatesVerbatim,
+      georeferenceSourcesText,
+      latitude,
+      longitude,
+      maximumDepthInMeters,
+      maximumElevationInMeters,
+      minimumDepthInMeters,
+      minimumElevationInMeters,
+      onEdit: handleEditCoordinate,
+      uncertaintyInMeters,
+    } = this.props
+
+    const hasDecimalCoordinates = latitude || longitude
+
+    let coordinates
```

Comment:
in this case I probably would've preferred a ternary...

```
const coordinates = hasDecimalCoordinates
    ? getCoordinates(latitude, longitude)
    : coordinatesVerbatim
```

I also know I said another time I preferred with `if` :-) I think the difference between the situations is that this situation requires a `let` and `else`. the `let` is always a bit worrying because you never know if that variable changes it's value again later in the code :-)

when the ternary can be replaced with if-then-return or return, it's less extra code and no added unclarity
```
if (something) {
  return a
}

return b
```
how do you think about it?